cdap-app-fabric/src/main/java/co/cask/cdap/runtime/monitor/RuntimeMonitor.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Runtime Monitor Runnable responsible for fetching metadata
+ */
+public class RuntimeMonitor extends AbstractExecutionThreadService {
+  private static final Logger LOG = LoggerFactory.getLogger(RuntimeMonitor.class);
+  private static final Gson GSON = new Gson();
+  private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() {}.getType();
+  private static final Type MAP_STRING_MESSAGE_TYPE = new TypeToken<Map<String, List<MonitorMessage>>>() {}.getType();
+
+  private final RESTClient restClient;
+  private final ClientConfig clientConfig;
+
+  private final int limit;
+  // map of configuration -> topic name
+  private Map<String, String> topicsToMonitor;
+  private Map<String, MonitorConsumeRequest> topicToRequest = new HashMap<>();
+  private String runtimeId;
+  private CConfiguration cConf;
+
+  private MessagePublisher messagePublisher;
+  private long pollTimeMs;
+  private volatile Thread runThread;
+  private volatile boolean stopped;
+
+  public RuntimeMonitor(String runtimeId, CConfiguration cConf, MessagePublisher messagePublisher,
+                        int pollTimeMs, ClientConfig clientConfig, int limit) {
+    this.runtimeId = runtimeId;
+    this.cConf = cConf;
+    this.messagePublisher = messagePublisher;
+    this.clientConfig = clientConfig;
+    this.restClient = new RESTClient(clientConfig);
+    this.limit = limit;
+    this.pollTimeMs = pollTimeMs;
+    this.topicsToMonitor = new HashMap<>();
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    LOG.debug("Starting runtime monitor {}....", runtimeId);
+    initializeTopics();
+  }
+
+  private void initializeTopics() throws Exception {
+    if (topicsToMonitor.isEmpty()) {
+      HttpResponse response = restClient.execute(HttpMethod.GET, clientConfig.getConnectionConfig().resolveURI
+                                                   (clientConfig.getApiVersion() +
+                                                      "/runtime/monitor/topics").toURL(),
+                                                 clientConfig.getAccessToken(),
+                                                 HttpURLConnection.HTTP_NOT_FOUND);
+      if (response.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        throw new NotFoundException(response.getResponseBodyAsString());
+      }
+
+      String topicsToMonitor = response.getResponseBodyAsString(Charset.forName("UTF-8"));
+      this.topicsToMonitor = GSON.fromJson(topicsToMonitor, MAP_STRING_STRING_TYPE);
+    }
+
+    // TODO initialize from offset table for a given runtimeId
+    for (String topic : topicsToMonitor.values()) {
+      topicToRequest.put(topic, new MonitorConsumeRequest(topic, null, limit));
+    }
+  }
+
+  @Override
+  protected void run() throws Exception {
+    runThread = Thread.currentThread();
+
+    try {
+      while (!stopped) {
+        HttpResponse response = restClient.execute(HttpMethod.GET, clientConfig.getConnectionConfig().resolveURI
```

Comment:
I missed in the API review, since you are submitting a request body, the request type should be `POST`